contracts/src/deployers/eeth/EETHHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,188 @@
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256) {
+        uint256 value;
+        // If base is the deposit asset, ensure that enough ether was sent to
+        // the contract and return the amount of ether that should be sent for
+        // the contribution.
+        if (_options.asBase) {
+            if (msg.value < _contribution) {
+                revert IHyperdriveDeployerCoordinator.InsufficientValue();
+            }
+            value = _contribution;
+        }
+        // Otherwise, transfer vault shares from the LP and approve the
+        // Hyperdrive pool.
+        else {
+            address token = _hyperdrive.vaultSharesToken();
+
+            // Convert the vault shares to base.
+            uint256 baseContribution = convertToBase(
+                IERC20(token),
+                _contribution
+            );
+
+            // NOTE: The eETH transferFrom function converts from base to shares under
+            // the hood using `sharesForAmount(_amount)`.
+            bool result = IeETH(token).transferFrom(
+                _lp,
+                address(this),
+                baseContribution
+            );
+            if (!result) {
+                revert IHyperdrive.TransferFailed();
+            }
+
+            IeETH(token).approve(address(_hyperdrive), baseContribution);
+        }
+        return value;
+    }
+
+    /// @notice Convert an amount of vault shares to an amount of base.
+    /// @param _vaultSharesToken The vault shares asset.
+    /// @param _shareAmount The vault shares amount.
+    /// @return The base amount.
+    function convertToBase(
+        IERC20 _vaultSharesToken,
+        uint256 _shareAmount
+    ) public view returns (uint256) {
+        return
+            EETHConversions.convertToBase(
+                liquidityPool,
+                _vaultSharesToken,
+                _shareAmount
+            );
+    }
+
+    /// @notice Convert an amount of base to an amount of vault shares.
+    /// @param _vaultSharesToken The vault shares asset.
+    /// @param _baseAmount The base amount.
+    /// @return The vault shares amount.
+    function convertToShares(
+        IERC20 _vaultSharesToken,
+        uint256 _baseAmount
+    ) public view returns (uint256) {
+        return
+            EETHConversions.convertToShares(
+                liquidityPool,
+                _vaultSharesToken,
+                _baseAmount
+            );
+    }
+
+    /// @dev We override the message value check since this integration is
+    ///      payable.
+    function _checkMessageValue() internal view override {}
+
+    /// @notice Checks the pool configuration to ensure that it is valid.
+    /// @param _deployConfig The deploy configuration of the Hyperdrive pool.
+    function _checkPoolConfig(
```

Comment:
We should also check that the base token is ETH and that the vault shares token is eETH. We can get the `eETH` address for this check by calling [`liquidityPool.eETH()`](https://github.com/etherfi-protocol/smart-contracts/blob/716fd013593c0a46f24b586db3849fa1d28cd989/src/LiquidityPool.sol#L37).